### PR TITLE
nxp: make it compatible with the lf-6.1.36-2.1.0 bsp release

### PR DESCRIPTION
**Pull request by ricardosalveti**

### CODE REVIEW

**Review by angolini — APPROVED** [review]
Everything looks ok to me. I haven't double-checked the default value of UBOOT_CONFIG for every imx, but I think it's aligned with what we need.

File: meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-bsp-common.inc [hunk 1]
```diff
@@ -11,14 +11,33 @@ deploy_nxp_imx() {
...
                     install -d ${DEPLOYDIR}/${BOOT_TOOLS}
-                    # When sign is enabled the final DTB should be copied from deploy dir
-                    if [ "${UBOOT_SIGN_ENABLE}" = "1" ]; then
-                        install -m 0644 ${DEPLOY_DIR_IMAGE}/${UBOOT_DTB_IMAGE} ${DEPLOYDIR}/${BOOT_TOOLS}/${UBOOT_DTB_NAME}
-                    else
-                        install -m 0644 ${B}/${config}/arch/arm/dts/${UBOOT_DTB_NAME} ${DEPLOYDIR}/${BOOT_TOOLS}
-                    fi
                     install -m 0644 ${B}/${config}/spl/u-boot-spl-nodtb.bin ${DEPLOYDIR}/${BOOT_TOOLS}/u-boot-spl-nodtb.bin-${MACHINE}-${type}
                     install -m 0644 ${B}/${config}/u-boot-nodtb.bin ${DEPLOYDIR}/${BOOT_TOOLS}/u-boot-nodtb.bin-${MACHINE}-${type}
+                    UBOOT_DTB_NAME_FLAGS="${type}:${UBOOT_DTB_NAME}"
```
Comment (angolini) [inline meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-bsp-common.inc:16]: s/u-boot-fio-bso-common/u-boot-fio-bsp-common in the commit title

File: meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-bsp-common.inc [hunk 2]
```diff
@@ -11,14 +11,33 @@ deploy_nxp_imx() {
...
+                            fi
+                            install -m 0644 ${B}/${config}/arch/arm/dts/${dtb_name}  ${DEPLOYDIR}/${BOOT_TOOLS}/${dtb_name}-${type}
+                        fi
+                        unset type_key
+                        unset dtb_name
+                    done
+
+                    unset UBOOT_DTB_NAME_FLAGS
                 fi
             done
```
Comment (angolini) [inline meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-bsp-common.inc:42]: So far for when UBOOT_CONFIG is empty, I've been using this:

```
+++ b/recipes-bsp/u-boot/u-boot-imx_2023.04.bb
@@ -53,6 +53,10 @@ do_deploy:append:mx8m-generic-bsp() {
             unset  j
         done
         unset  i
+    else
+        install -d ${DEPLOYDIR}/${BOOT_TOOLS}
+        install -m 0644 ${B}/${config}/arch/arm/dts/${UBOOT_DTB_NAME}  ${DEPLOYDIR}/${BOOT_TOOLS}/${UBOOT_DTB_NAME}
+        install -m 0644 ${B}/${config}/u-boot-nodtb.bin                ${DEPLOYDIR}/${BOOT_TOOLS}/u-boot-nodtb.bin-${MACHINE}
     fi
 }
```

Comment (ricardosalveti) [reply meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-bsp-common.inc:42]: Cool, we can follow this up with your other changes in meta-freescale.

**Pull request merged**